src/plugins/ui_actions/public/service/ui_actions_service.ts

```diff
@@ -138,7 +139,9 @@ export class UiActionsService {
     triggerId: TType,
     // The action can accept partial or no context, but if it needs context not provided
     // by this type of trigger, typescript will complain. yay!
-    action: ActionByType<AType> & Action<TriggerContextMapping[TType]>
+    action:
+      | (ActionByType<AType> & Action<TriggerContextMapping[TType]>)
+      | ActionDefinition<ActionContextMapping[AType]>
```

Comment:
I think this change broke the type checking.

On master:

<img width="1084" alt="Screen Shot 2020-04-14 at 6 03 57 PM" src="https://user-images.githubusercontent.com/16563603/79278763-65f6bb80-7e7a-11ea-9893-149a5332d01b.png">

With this PR:

<img width="1002" alt="Screen Shot 2020-04-14 at 6 04 54 PM" src="https://user-images.githubusercontent.com/16563603/79278796-77d85e80-7e7a-11ea-80a4-f447b24a033b.png">

I soooo wish we had a whole Kibana system that includes typescripts tests

Reply:
>I soooo wish we had a whole Kibana system that includes typescripts tests

I suspect this would be avoided by factoring out the `id` portion of the function signature and registering the action directly to the trigger.

I think its possible to write tests against typescript types but you'd need to use compiler functionality. Unlikely to be worth it, although I'd be super interested to find a case that calls for it.

Reply:
I've reverted this change.

Reply:
> I think its possible to write tests against typescript types but you'd need to use compiler functionality. Unlikely to be worth it, although I'd be super interested to find a case that calls for it.

The canvas team actually does this!  It's just not part of the entire Kibana system yet.  

This is the only issue I could find about making this more official: https://github.com/elastic/kibana/issues/53762

I'm looking for where actually in Canvas code these tests are but I can't.  Will report back.